src/ai/susi/server/api/aaa/GetAvatarServlet.java

```diff
@@ -0,0 +1,86 @@
+/**
+ *  GetAvatarService
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program in the file lgpl21.txt
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package ai.susi.server.api.aaa;
+
+import ai.susi.DAO;
+import ai.susi.json.JsonObjectWithDefault;
+import ai.susi.mind.SusiSkill;
+import ai.susi.server.*;
+import org.json.JSONObject;
+import org.apache.commons.io.FileUtils;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Map;
+
+import java.nio.file.Files;
+import ai.susi.server.Query;
+import ai.susi.server.RemoteAccess;
+import javax.servlet.ServletException;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+
+/*
+ This Servlet gives a API Endpoint to return avatar of a user
+ http://localhost:4000/getAvatar.png?image=963e84467b92c0916b27d157b1d45328.png
+ */
+
+ public class GetAvatarServlet extends HttpServlet {
+
+	private static final long serialVersionUID = 7408901113114682419L;
+
+	@Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        Query post = RemoteAccess.evaluate(request);
+        File imageFile = null;
+        String file = "";
+
+        if (post.get("image","") != "") {
```

Comment:
I don't think that it is good that the client has to know the image file or path. The client should just say "give me the users avatar". You can use the user's credential to extract the Uuid of the user. That is used to store the image according to my latest comment in the corresponding pull request